docs/src/dev/provider/index.asciidoc

```diff
@@ -848,6 +848,14 @@ Server returns for a single request.  Again, this description of Gremlin Server'
 out-of-the-box configuration.  It is quite possible to construct other flows, that might be more amenable to a
 particular language or style of processing.
 
+It is recommended but not required that a driver include a `User-Agent` header as part of any web socket
```

Comment:
What are the side effects of enabling it? We should help the users make this decision. Please add a jmh benchmark for handshake w/ and w/o the user agent. User agent will add additional latency to the handshake and hence, connection setup will be slower but by how much?

Reply:
How much value do you see in having such a benchmark? My understanding is that the jmh benchmarks aren't currently setup for measuring anything like the driver handshake latency. I also have concerns about how usefulness of any result as the latency will be highly dependent on network conditions. A typical user agent in the format we are using is under 100 bytes and it is only sent once per connection in the handshake. Personally I don't foresee there being any noticeable side effects other than the obvious added logging and metrics.

Some sort of benchmark can certainty be added but I don't believe it will be trivial to add and I worry it won't offer much value for the investment put into creating it.

Reply:
There are users of TinkerPop which open one connection per request and hence, they are sensitive to handshake latency. Theoretically the latency incurred will be due to increased size of header (hence longer header parsing time from bytebuffer to Java object + longer time to send larger object over network). 

If it is not trivial to calculate this, then we can ignore this since I believe there are larger bottlenecks in handshake than header parsing right now.

Reply:
I would like to ignore this for now. It seems that designing a good benchmark would take a reasonable amount of time and it is not clear to me how I would normalize results to account for various network conditions.